### DISCUSSION TITLE
Not understanding how to use Mojo::IOLoop::Subprocess

### DISCUSSION THREAD

**Discussion by john-** [mojolicious/mojo#1744] [Q&A] [answered]
I have not used Subprocess before so probably missing something basic.   In this code snippet I would expect to see debug message "DOES NOT GET HERE":
```
  my $cb = $c->pg->pubsub->listen(audio => sub {
        my ($pubsub, $payload) = @_;
        my $msg = decode_json($payload);
        app->log->debug("GETS HERE.  msg: " . Dumper($msg));
        my $subprocess = Mojo::IOLoop->Subprocess->new;
        $subprocess->run(
            sub ($subprocess) {
                app->log->debug('DOES NOT GET HERE');
                $msg = $watcher->process($msg);  # store in DB, add additional fields, etc                     
                return $msg;
            },
            sub ($subprocess, $err, @results) {
                app->log->debug("got results: $results[0]");
                #app->log->debug("item to let client know about (pubsub): $msg->{type}");                      
                #$c = $c->send( { json => $msg } );                                                            
            }
        );
    });
```

**Comment by Grinnz** [comment, marked as answer]
I assume `Mojo::IOLoop->Subprocess` is a typo. Regardless you should create it using `Mojo::IOLoop->subprocess(sub {}, sub {})`, or better yet `Mojo::IOLoop->subprocess->run_p`: https://metacpan.org/pod/Mojo::IOLoop#subprocess

Context is also important. This seems like it's in Mojolicious application context so the IOLoop should already be running. If it was not, you would have to run it in order for the subprocess to be executed.

**Reply by john-** [reply]
I corrected the typo and it now seems to work as expected.   I will take into account your other suggestions as well.
